filesystem/fs_interface.go

```diff
@@ -0,0 +1,14 @@
+package filesystem
+
+type Client interface {
+	Open() File
```

Comment:
Sure. I think OpenWriteCloser() can create the file if not existed. Should be optional. This is enough for current need.